mkosi/distributions/debian.py

```diff
@@ -224,7 +227,27 @@ def invoke_apt(
         INITRD="No",
     )
 
-    return bwrap(["apt-get", operation, *extra], apivfs=state.root if apivfs else None, env=env | state.environment)
+    with ExitStack() as stack:
+        extra_apt_keyrings = state.root / "etc/apt/keyrings"
+        extra_usr_keyrings = state.root / "usr/share/keyrings"
+
+        if extra_apt_keyrings.exists():
+            stack.enter_context(mount(extra_apt_keyrings, Path("/etc/apt/keyrings"), options=['bind']))
+
+        if extra_usr_keyrings.exists():
+            keydir = stack.enter_context(tempfile.TemporaryDirectory(prefix="usr-share-keyrings-", dir=state.workspace))
+
+            # Copy the contents of the host keyring to prevent bootstrap issues
+            shutil.copytree("/usr/share/keyrings", keydir, dirs_exist_ok=True)
+
+            # Copy the additonal keyrings
+            shutil.copytree(extra_usr_keyrings, keydir, dirs_exist_ok=True)
+
+            # Create a bind mount using the combined key directory
+            # OverlayFS is not used because it is unknown which partitions data resides on
+            stack.enter_context(mount(keydir, Path("/usr/share/keyrings"), options=['bind']))
+
+        return bwrap(["apt-get", operation, *extra], apivfs=state.root if apivfs else None, env=env | state.environment)
```

Comment:
No need to muck around with bind mounts, we have full control over where apt searches for the keyring via its config parameters `Dir::etc::trusted` and `Dir::etc::trustedparts`.

Reply:
Maybe not. Take a look (here)[https://github.com/systemd/mkosi/pull/1490#issuecomment-1524984015]. I've added the details merely for clarity here. This PR is likely to get closed.